### PR TITLE
Target netcoreapp3.1 instead of 2.0 for KeyDeriviation DataProtection

### LINKED ISSUE #14023: Investigate TFMs of Microsoft.AspNetCore.Cryptography.KeyDerivation
Tracking the investigation that started in https://github.com/aspnet/AspNetCore/pull/13571#discussion_r324443496. The package currently targets netcoreapp2.0 and netstandard2.0. We should double check if netcoreapp2.0 TFM is intentional.

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/AspNetCore/issues/14023

### CODE REVIEW

File: src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/NetCorePbkdf2Provider.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-#if NETCOREAPP2_0
+#if NETCOREAPP3_1
```
Comment (JunTaoLuo) [inline src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/NetCorePbkdf2Provider.cs]:
```suggestion
#if NETCOREAPP
```

File: src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/Pbkdf2Util.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ private static IPbkdf2Provider GetPbkdf2Provider()
             {
 #if NETSTANDARD2_0
                 return new ManagedPbkdf2Provider();
-#elif NETCOREAPP2_0
+#elif NETCOREAPP3_1
```
Comment (JunTaoLuo) [inline src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/Pbkdf2Util.cs]:
```suggestion
#elif NETCOREAPP
```

**Review by JunTaoLuo — APPROVED** [review]

**Comment by HaoK** [comment]
Retargetted to master

**Comment by HaoK** [comment]
Replaced by https://github.com/aspnet/AspNetCore/pull/14633 against master

**Pull request closed without merging**